### PR TITLE
fix: concat for bytesM, M<32

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
fix concat for `bytesM` where `M < 32` (new bytes types).

### How I did it
fix the codegen, also use routines from vyper.codegen.core.

h/t to @hedgar2017 for reporting

### How to verify it
see tests

### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../blob/master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

image

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: tests/parser/functions/test_concat.py [hunk 1]
```diff
@@ -121,6 +123,56 @@ def small_output(a: String[5], b: String[4]) -> String[9]:
...
+
+@external
+def small_bytes4(a: bytes8, b: Bytes[32], c: bytes8) -> Bytes[48]:
+    return concat(a, b, c)
+    """
+    contract = get_contract_with_gas_estimation(code)
+
+    def bytes_for_len(n):
+        # note: python 3.9 has `randbytes` for this.
+        return random.randrange(2 ** (8 * n)).to_bytes(n, "big")
```
Comment (fubuloubu) [inline tests/parser/functions/test_concat.py:149]: Can't you test values for bytes that are edge cases of behavior? My problem with introducing non-determinism into the test suite like this is that there is not an easy way to replicate a failure by setting the seed like you can in hypothesis.

Comment (charles-cooper) [reply tests/parser/functions/test_concat.py:149]: how about if we set and report the random seed used? https://github.com/charles-cooper/vyper/commit/77c2b1d11f1582921d82f94c058d360d74ca0b90

Comment (charles-cooper) [reply tests/parser/functions/test_concat.py]: @fubuloubu resolved in https://github.com/vyperlang/vyper/pull/2764/commits/0fa645c25f3f56549b0660d013d9debb79c7780c

Comment (charles-cooper) [reply tests/parser/functions/test_concat.py]: actually -- see https://github.com/vyperlang/vyper/pull/2764/commits/e8ad0508740c03645e7fcb7c4fd516cb2ffdb6e7

**Review by fubuloubu — APPROVED** [review]
